Tensile/Common.py

```diff
@@ -2495,7 +2500,8 @@ def assignGlobalParameters( config ):
   if "IgnoreAsmCapCache" in config:
     globalParameters["IgnoreAsmCapCache"] = config["IgnoreAsmCapCache"]
     
-  globalParameters["AsmCaps"] = {}
+  globalParameters["CacheAsmCaps"] = True if capabilitiesCache is not None else False
```

Comment:
From the way I understand this change, this will cache an empty dict if the cache file doesn't exist on disk since `{}` will be returned, which is not none. Is this the intended behaviour?

Reply:
If a user provides an asm cache file, but that file doesn't exists that means I want a cache but I don't yet have one (i.e. a clean run of Tensile). So when you reach this point in the code `capabilitiesCache` would be `{}` and `bool(capabilitiesCache0 == False` so we would return false even though we really want to cache the asm caps. That is why we set it to None for the case where we do not want a cache and check for `None` i.e. an it is an `Optional[Dict]`.  If the optional is engaged that means we want a cache if not we don't. Maybe there is a more pythonic way of checking if an optional is engaged.